### DISCUSSION TITLE
"gff" and "bed" tracks display coloring modification

### DISCUSSION THREAD

**Discussion by Alber-Tadrous** [GMOD/jbrowse-components#3547] [Q&A]
I am working on project of creating a processor to configure jbrowse instance through cli automatically using the metadata attached to each assembly and track file and one of the features we want to add is to configure the styling of bigwig, gff and bed tracks "coloring for starter", but I failed to find out how could I configure the display types used in by gff and bed track. Would you please point me to any sort of docs to do so, if it exists?

**Comment by cmdcolin** [comment]
just to preface, configuring simple things like color can admittedly be a little tricky. The basic premise to configuring color involves

track config->display config->renderer

the display is the configuration for displaying a track in a particular view type (e.g. linear or circular) and the renderer is a thing called by the display to render the data, and there can be multiple types of renderers for a given display type (bigwig has density, xyplot, and line renderers)

so, to configure the color for a bigwig track to use XYPlot as pink, it looks like this

```json
{
      "type": "QuantitativeTrack",
      "trackId": "bigwigtrack",
      "name": "BigWig file (custom color)",
      "assemblyNames": ["volvox"],
      "adapter": {
        "type": "BigWigAdapter",
        "bigWigLocation": {  "uri": "volvox_microarray.bw"  }
      },
      "displays": [
        {
          "type": "LinearWiggleDisplay",
          "displayId": "volvox_microarray_xyplot",
          "defaultRendering": "xyplot",
          "renderers": {
            "XYPlotRenderer": { "color": "pink" }
          }
        }
      ]
    },
```

for a BED track or GFF (for GFF, just swap out the adapter)

```json
{
      "type": "FeatureTrack",
      "trackId": "bedtrack",
      "name": "BED file (custom color)",
      "assemblyNames": ["volvox"],
      "adapter": {
        "type": "BedAdapter",
        "bedLocation": {
          "uri": "myfile.bed"
        }
      },
      "displays": [
        {
          "type": "LinearBasicDisplay",
          "displayId": "bedtrack-LinearBasicDisplay",
          "renderer": {
            "type": "SvgFeatureRenderer",
            "color1": "purple"
          }
        }
      ]
    },

```

there is an example of loading color via the CLI here https://jbrowse.org/jb2/docs/faq/#adding-color-callbacks-via-the-command-line, but essentially, you just want to generate JSON that looks like the above

potentially in the future, we may try to find a more streamlined way to configure simple things like color, issue for this here https://github.com/GMOD/jbrowse-components/issues/3424

**Reply by cmdcolin** [reply]
note: displayId is generally just any uniquely identifiable string, you can randomly generate it but by convention it is normally trackId+the display type
